metadata-ingestion/src/datahub/ingestion/source/mode.py

```diff
@@ -1469,16 +1483,21 @@ def _get_request_json(self, url: str) -> Dict:
                 multiplier=self.config.api_options.retry_backoff_multiplier,
                 max=self.config.api_options.max_retry_interval,
             ),
-            retry=retry_if_exception_type(HTTPError429),
+            retry=retry_if_exception_type((HTTPError429, ConnectionError)),
             stop=stop_after_attempt(self.config.api_options.max_attempts),
         )
 
         @r.wraps
         def get_request():
             try:
-                response = self.session.get(url)
-                response.raise_for_status()
+                response = self.session.get(
+                    url, timeout=self.config.api_options.timeout
+                )
                 return response.json()
+            except ConnectionError as e:
+                # TODO Need to check why server is closes the connection before sending a complete response
+                logging.warning(f"RemoteDisconnected or ProtocolError: {str(e)}")
+                return {}
```

Comment:
if we don't get any info from the API, does it really make sense to pretend like the API returned nothing?

Reply:
Idea is to debug, dont need to pretend here. Removed this !!!